frontend/src/pages/projects/screens/spawner/storage/useAvailablePvcs.ts

```diff
@@ -13,15 +17,20 @@ const useAvailablePvcs = (
     if (projectName) {
       getPvcs(projectName)
         .then((newPvcs) => {
-          setPvcs(newPvcs);
+          let usedPvcs: string[] = [];
+          notebooks.forEach((notebook) => {
+            const pvcNames = getNotebookPVCNames(notebook.notebook);
+            usedPvcs = _.union(usedPvcs, pvcNames);
```

Comment:
I'm not a fan of this -- but I guess we really don't have time to mess with it. Try to avoid mutating references and just use `.forEach` for side-effect actions -- `.map` is what you wanted here.

Reply:
I think what you wanted was .map + _.flat + _.uniq

This code works, but it is just not great for maintenance